### PR TITLE
Task/new update endpoints

### PR DESCRIPTION
**Pull request by krisorr**
# What this does
Adds new explicit endpoints for updating checkout addresses, and shipping rate.

Sending the whole checkout on patch operations has been causing issues with validation on the back end.

# Issues
#142 

@yervantk @sav007 please review

### CODE REVIEW

File: MobileBuy/buy/src/main/java/com/shopify/buy/dataprovider/CheckoutServiceDefault.java [hunk 1]
```diff
@@ -134,12 +135,47 @@ public CancellableTask createCheckout(final Checkout checkout, final Callback<Ch
...
+            throw new NullPointerException("checkout token cannot be empty");
+        }
+
+        Checkout checkout = new Checkout(checkoutToken);
+        checkout.setShippingRate(shippingRate);
+
+        return updateCheckout(checkout);
+    }
+
+    @Override
```
Comment (sav007) [inline MobileBuy/buy/src/main/java/com/shopify/buy/dataprovider/CheckoutServiceDefault.java]: Shouldn't it be marked as deprecated?

File: MobileBuy/buy/src/main/java/com/shopify/buy/dataprovider/CheckoutServiceDefault.java [hunk 2]
```diff
@@ -134,12 +135,47 @@ public CancellableTask createCheckout(final Checkout checkout, final Callback<Ch
...
     public CancellableTask updateCheckout(final Checkout checkout, final Callback<Checkout> callback) {
         return new CancellableTaskSubscriptionWrapper(updateCheckout(checkout).subscribe(new InternalCallbackSubscriber<>(callback)));
     }
 
     @Override
-    public Observable<Checkout> updateCheckout(final Checkout checkout) {
+    public Observable<Checkout> updateCheckout(Checkout checkout) {
```
Comment (sav007) [inline MobileBuy/buy/src/main/java/com/shopify/buy/dataprovider/CheckoutServiceDefault.java]: Shouldn't it be marked as deprecated?

Comment (krisorr) [inline MobileBuy/buy/src/main/java/com/shopify/buy/dataprovider/CheckoutServiceDefault.java]: no, we still need it in other places

File: MobileBuy/buy/src/main/java/com/shopify/buy/dataprovider/CheckoutServiceDefault.java (hunk 1 above)
Comment (sav007) [inline MobileBuy/buy/src/main/java/com/shopify/buy/dataprovider/CheckoutServiceDefault.java]: Shouldn't it be marked as deprecated?

**Comment by sav007** [comment]
LGTM

**Pull request merged**